# Rosalind Bioinformatics Stronghold
:rocket: Solutions for the Bioinformatics Stronghold Problems from Rosalind 

1. [Counting DNA Nucleotides](solutions/dna.py)
1. [Transcribing DNA into RNA](solutions/rna.py)
1. [Complementing a Strand of DNA](solutions/revc.py)
1. [Mendel's First Law](solutions/iprb.py)
1. [Rabbits and Recurrence Relations](solutions/fib.py)
1. [Computing GC Content](solutions/gc.py)
1. [Translating RNA into Protein](solutions/prot.py)
1. [Finding a Motif in DNA](solutions/subs.py)
1. [Counting Point Mutations](solutions/hamm.py)
1. [Calculating Expected Offspring](solutions/iev.py)
1. [Mortal Fibonacci Rabbits](solutions/fibd.py)
1. [Inferring mRNA from Protein](solutions/mrna.py)
1. [Independent Alleles](solutions/lia.py)
1. [Calculating Protein Mass](solutions/prtm.py)
1. [Overlap Graphs](solutions/grph.py)
1. [Finding a Protein Motif](solutions/mprt.py)
1. [Consensus and Profile](solutions/cons.py)
1. [Open Reading Frames](solutions/orf.py)
1. [RNA Splicing](solutions/splc.py)
1. [Finding a Shared Motif](solutions/lcsm.py)
1. [Enumerating Gene Orders](solutions/perm.py)
1. [Locating Restriction Sites](solutions/revp.py)
1. [Perfect Matchings and RNA Secondary Structures](solutions/pmch.py)
1. [Partial Permutations](solutions/pper.py)
1. [Completing a Tree](solutions/tree.py)
1. [Introduction to Random Strings](solutions/prob.py)
1. [Genome Assembly as Shortest Superstring](solutions/long.py)
1. [Finding a Spliced Motif](solutions/sseq.py)
1. [Transitions and Transversions](solutions/tran.py)
1. [Enumerating k-mers Lexicographically](solutions/lexf.py)
1. [Longest Increasing Subsequence](solutions/lgis.py)
1. [Enumerating Oriented Gene Orderings](solutions/sign.py)
1. [Maximum Matchings and RNA Secondary Structures](solutions/mmch.py)
1. [Counting Phylogenetic Ancestors](solutions/inod.py)
1. [Counting Subsets](solutions/sset.py)
1. [Catalan Numbers and RNA Secondary Structures](solutions/cat.py)
1. [Speeding Up Motif Finding](solutions/kmp.py)
1. [Creating a Distance Matrix](solutions/pdst.py)
1. [Matching Random Motifs](solutions/rstr.py)
1. [k-Mer Composition](solutions/kmer.py)
1. [Error Correction in Reads](solutions/corr.py)
1. [Finding a Shared Spliced Motif](solutions/lcsq.py)
1. [Ordering Strings of Varying Length Lexicographically](solutions/lexv.py)
1. [Reversal Distance](solutions/rear.py)
1. [Distances in Trees](solutions/nwck.py)
1. [Introduction to Alternative Splicing](solutions/aspc.py)
1. [Introduction to Set Operations](solutions/seto.py)
1. [Motzkin Numbers and RNA Secondary Structures](solutions/motz.py)
1. [Inferring Protein from Spectrum](solutions/spec.py)
1. [Introduction to Pattern Matching](solutions/trie.py)
1. [Interleaving Two Motifs](solutions/scsp.py)
1. [Expected Number of Restriction Sites](solutions/eval.py)
1. [Edit Distance](solutions/edit.py)
1. [Sorting by Reversals](solutions/sort.py)
1. [Newick Format with Edge Weights](solutions/nkew.py)
1. [Creating a Character Table](solutions/ctbl.py)
1. [Independent Segregation of Chromosomes](solutions/indc.py)
1. [Constructing a De Bruijn Graph](solutions/dbru.py)
1. [Comparing Spectra with the Spectral Convolution](solutions/conv.py)
1. [Wobble Bonding and RNA Secondary Structures](solutions/rnas.py)
1. [Inferring Peptide from Full Spectrum](solutions/full.py)
1. [Finding Disjoint Motifs in a Gene](solutions/itwv.py)
1. [Finding the Longest Multiple Repeat](solutions/lrep.py)
1. [Edit Distance Alignment](solutions/edta.py)
1. [Counting Unrooted Binary Trees](solutions/cunr.py)
1. [Quartets](solutions/qrt.py)
1. [Counting Disease Carriers](solutions/afrq.py)
1. [Creating a Character Table from Genetic Strings](solutions/cstr.py)
1. [Genome Assembly with Perfect Coverage](solutions/pcov.py)
1. [Matching a Spectrum to a Protein](solutions/prsm.py)
1. [Using the Spectrum Graph to Infer Peptides](solutions/sgra.py)
1. [Encoding Suffix Trees](solutions/suff.py)
1. [Counting Optimal Alignments](solutions/ctea.py)
1. [Global Alignment with Scoring Matrix](solutions/glob.py)
1. [Enumerating Unrooted Binary Trees](solutions/eubt.py)
1. [Counting Rooted Binary Trees](solutions/root.py)
1. [Phylogeny Comparison with Split Distance](solutions/sptd.py)
1. [Counting Quartets](solutions/cntq.py)
1. [Inferring Genotype from a Pedigree](solutions/mend.py)
1. [The Wright-Fisher Model of Genetic Drift](solutions/wfmd.py)
1. [Sex-Linked Inheritance](solutions/sexl.py)
1. [Character-Based Phylogeny](solutions/chbp.py)
1. [Genome Assembly Using Reads](solutions/gasm.py)
1. [Creating a Restriction Map](solutions/pdpl.py)
1. [Identifying Maximal Repeats](solutions/mrep.py)
1. [Multiple Alignment](solutions/mult.py)
1. [Linguistic Complexity of a Genome](solutions/ling.py)
1. [Maximizing the Gap Symbols of an Optimal Alignment](solutions/mgap.py)
1. [Local Alignment with Scoring Matrix](solutions/loca.py)
1. [Global Alignment with Constant Gap Penalty](solutions/gcon.py)

1. [Assessing Assembly Quality with N50 and N75](solutions/asmq.py)

1. [Overlap Alignment](solutions/oap.py)
1. [Finding a Motif with Modifications](solutions/sims.py)
1. [Semiglobal Alignment](solutions/smgb.py)
1. [Global Alignment with Scoring Matrix and Affine Gap Penalty](solutions/gaff.py)


1. [Local Alignment with Affine Gap Penalty](solutions/laff.py)
